modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/jaxrs/application/FeatureFlagApplication.java

```diff
@@ -134,8 +137,8 @@ private List<FeatureFlag> _getDependencyFeatureFlags(
 		if (featureFlag == null) {
 			_log.error(
 				StringBundler.concat(
-					"Feature flag ", key, " does not exist for company ",
-					companyId));
+					"Feature flag ", HtmlUtil.escape(key),
+					" does not exist for company ", companyId));
```

Comment:
Hey @pat270 Since this won't be directly placed in html I think we would want to use a different escape method. I think we would do this instead:
```
String.format("Feature flag %s does not exist for company %s", key, companyId)
```
String.format automatically escapes the variables that subbed in.

I think we would use the same method for escaping the response above, but let's get this verified by our backend devs. @thiago-buarqque @algobob what's the right way to handle this escape?

Reply:
From the [Java String docs](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#format-java.lang.String-java.lang.Object...-:~:text=by%20this%20string.-,format,-public%20static%C2%A0), it seems that `String.format` generates the same result as the `StringBundle.concat`. Since it doesn't mention escaping anything. 

The error message is not rendered on the UI, we just use it to create some tests (`modules/test/playwright/fixtures/featureFlagsTest.ts`) and a helper (`modules/test/playwright/helpers/FeatureFlagApiHelper.ts`).

Reply:
If we still have to escape the message, the HTML escape will be probably ideal. I say that because I can't imagine another escaping for this case

Reply:
So was this reported in error since it doesn't get output to the UI?

Reply:
Actually, now I see there is an issue regarding the response. Even though an error message is sent on the response body, the http code will be 200. 

I'll check with Drew Brokke regarding this, give me some time

Reply:
@pat270 , I checked and turns out Ivan Zaera added this without consulting us. 

[Ticket](https://liferay.atlassian.net/browse/LPD-5118)
[Commit](https://github.com/liferay/liferay-portal/commit/9c03a3a03399067f89ad5c875e67453d574dda01)

I propose we go with the HTML escape and, in the near future, we:

- Update the response code from 200 to 404 in case the FF is not found and to 500 in case there is an exception;
- Remove the error message, it's not sent to the user anyway. We could just rely on the HTTP code.


If @ethib137 gives a thumbs up, you can run CI and go with the HTML escape approach.

Reply:
@thiago-buarqque can you create a ticket for the additional changes that we need to make so that we can track it.

Reply:
Created task https://liferay.atlassian.net/browse/LPD-19915